core/src/main/java/hudson/model/DirectoryBrowserSupport.java

```diff
@@ -183,9 +184,13 @@ private void serveFile(StaplerRequest req, StaplerResponse rsp, VirtualFile root
             while(pathTokens.hasMoreTokens()) {
                 String pathElement = pathTokens.nextToken();
                 // Treat * and ? as wildcard unless they match a literal filename
-                if((pathElement.contains("?") || pathElement.contains("*"))
-                        && inBase && !root.child((_base.length() > 0 ? _base + "/" : "") + pathElement).exists())
-                    inBase = false;
+                try {
+                    if((pathElement.contains("?") || pathElement.contains("*"))
+                            && inBase && !root.child((_base.length() > 0 ? _base + "/" : "") + pathElement).exists())
+                        inBase = false;
+                } catch (InvalidPathException e) {
+                    // * is an illegal filename component on some systems, see JENKINS-31015 -- ignore
```

Comment:
should this not return an error page or at least break the while loop or something here?